src/main/java/duke/command/Storage.java

```diff
@@ -0,0 +1,131 @@
+package duke.command;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import static duke.command.TaskList.printExceptionMessage;
+
+/**
+ * Deals with loading tasks from the file and saving task
+ * in the file
+ */
+public class Storage {
+
+    private static final String ERROR_DATE_TIME_FORMAT = "Incorrect date/time format provided. Use YYYY-MM-DD,HH:MM.";
+    private static final String ERROR_READING_FILE = "Unable to read file. Check the text file again";
+    private final String SEGMENT_DIVIDER = " | ";
+    private String file;
+
+    public Storage(String file) {
+        setFile(file);
+    }
+
+    public String getFile() {
+        return file;
+    }
+
+    public void setFile(String file) {
+        this.file = file;
+    }
+
+    /**
+     * Writes file in a specific format with individual segments
+     * of the task in the taskList
+     *
+     * @param taskList the list that stores the current tasks
+     * @throws IOException exception with inputs and outputs with writing
+     */
+    public void saveFile(ArrayList<Task> taskList) throws IOException {
+
+        FileWriter fw = new FileWriter(file);
+        String inputs = "";
+        for (Task task: taskList) {
+            inputs += task.getTaskType();
+            inputs += SEGMENT_DIVIDER;
+            inputs += task.isDoneChecker() ? "1" : "0";
+            inputs += SEGMENT_DIVIDER;
+            inputs += task.getTaskName();
+            inputs += SEGMENT_DIVIDER;
+            inputs += task.getTaskDetails();
+            inputs += "\n";
+        }
+        fw.write(inputs);
+        fw.close();
+    }
+
+    /**
+     * Loads the text file and extracts relevant information
+     * to recover the previously saved task list
+     *
+     * @return taskList the list that stores the current tasks
+     * @throws FileNotFoundException if file cant be read
+     */
+    public ArrayList<Task> loadFile() throws FileNotFoundException {
```

Comment:
Try breaking this method into bunch of smaller methods (as a general rule, avoid writing methods that are more than 30 lines)